Core/GDCore/Project/Layer.h

```diff
@@ -219,16 +229,18 @@ class GD_CORE_API Layer {
   void UnserializeFrom(const SerializerElement& element);
 
  private:
-  gd::String name;                  ///< The name of the layer
-  bool isVisible;                   ///< True if the layer is visible
-  bool isLightingLayer;             ///< True if the layer is used to display lights and renders an ambient light.
-  bool followBaseLayerCamera;           ///< True if the layer automatically follows the base layer
-  unsigned int ambientLightColorR;     ///< Ambient light color Red component
-  unsigned int ambientLightColorG;     ///< Ambient light color Green component
-  unsigned int ambientLightColorB;     ///< Ambient light color Blue component
+  gd::String name;       ///< The name of the layer
+  bool isVisible;        ///< True if the layer is visible
+  bool isLightingLayer;  ///< True if the layer is used to display lights and
+                         ///< renders an ambient light.
+  bool followBaseLayerCamera;  ///< True if the layer automatically follows the
+                               ///< base layer
+  unsigned int ambientLightColorR;  ///< Ambient light color Red component
+  unsigned int ambientLightColorG;  ///< Ambient light color Green component
+  unsigned int ambientLightColorB;  ///< Ambient light color Blue component
   std::vector<gd::Camera> cameras;  ///< The camera displayed by the layer
-  std::vector<std::shared_ptr<gd::Effect>>
-      effects;  ///< The effects applied to the layer.
+  std::unique_ptr<gd::EffectsContainer>

```

Comment:
No need for a unique_ptr here. Just use a `gd::EffectsContainer effectsContainer; ///< Th effects applied to the layer`.
By using this, the compiler will properly copy the effects container for you when a layer is copied, it will initialize it, and you don't need to worry about the copy constructor too.